contract_comm/blockchain_parameters/blockchain_parameters.go

```diff
@@ -83,6 +127,49 @@ func GetMinimumVersion(header *types.Header, state vm.StateDB) (*params.VersionI
 	return &params.VersionInfo{version[0].Uint64(), version[1].Uint64(), version[2].Uint64()}, nil
 }
 
+func GetGasCost(header *types.Header, state vm.StateDB, defaultGas uint64, method string) uint64 {
+	var gas *big.Int
+	var err error
+	_, err = contract_comm.MakeStaticCall(
+		params.BlockchainParametersRegistryId,
+		blockchainParametersABI,
+		method,
+		[]interface{}{},
+		&gas,
+		params.MaxGasForReadBlockchainParameter,
+		header,
+		state,
+	)
+	if err != nil {
+		log.Info("Default gas", "gas", defaultGas, "method", method)
+		return defaultGas
+	}
+	log.Info("Reading gas", "gas", gas)
+	return gas.Uint64()
+}
+
+func GetGasForDebitFromTransactions(header *types.Header, state vm.StateDB) uint64 {
+	return GetGasCost(header, state, params.ExpectedGasForDebitFromTransactions, "gasForDebitFromTransactions")
+}
+
+func GetGasForCreditToTransactions(header *types.Header, state vm.StateDB) uint64 {
+	return GetGasCost(header, state, params.ExpectedGasForCreditToTransactions, "gasForCreditToTransactions")
+}
+
+func GetGasToReadErc20Balance(header *types.Header, state vm.StateDB) uint64 {
+	return GetGasCost(header, state, params.ExpectedGasToReadErc20Balance, "gasToReadErc20Balance")
+}
+
+func GetIntrinsicGasForAlternateGasCurrency(header *types.Header, state vm.StateDB, gasCurrency *common.Address) uint64 {
+	if gasCurrency == nil {
```

Comment:
Can we do this check inside `IntrinsicGas` instead? Otherwise, would recommend renaming this function to `GetIntrinsicGasForGasCurrency`